test/test_jit.py

```diff
@@ -13796,6 +13796,11 @@ def test_non_primitive_types(x):
         out = test_non_primitive_types(_MyNamedTuple(value=torch.tensor(5.0)))
         self.assertEqual(out, torch.tensor(6.0))
 
+        with self.assertRaisesRegex(RuntimeError, 'inferred'):
```

Comment:
Though we are indeed trying to check that tensor type has `inferred` set to true, it is better if we check more than just `inferred` in the error message.

Also, it is generally better to minimize amount of code under `self.assertRaisesRegex`, so I would suggest define `test_error` outside and only call `torch.jit.script` on it inside `self.assertRaisesRegex` scope.

Reply:
Got it!